src/validation.cpp

```diff
@@ -3803,10 +3803,13 @@ bool LoadChainTip(const CChainParams& chainparams)
 
     g_chainstate.PruneBlockIndexCandidates();
 
-    LogPrintf("Loaded best chain: hashBestChain=%s height=%d date=%s progress=%f\n",
-        chainActive.Tip()->GetBlockHash().ToString(), chainActive.Height(),
-        DateTimeStrFormat("%Y-%m-%d %H:%M:%S", chainActive.Tip()->GetBlockTime()),
-        GuessVerificationProgress(chainparams.TxData(), chainActive.Tip()));
+    {
+        LOCK(cs_main);
```

Comment:
The whole function should lock `cs_main`. However this is called only called in `AppInitMain`, and #11041 adds the lock [there](https://github.com/bitcoin/bitcoin/pull/11041/files#diff-c865a8939105e6350a50af02766291b7).

If this goes first, I suggest cherry pick 3d40869 instead.